android/src/main/java/com/swmansion/reanimated/keyboard/ModalActivityManager.java

```diff
@@ -0,0 +1,108 @@
+package com.swmansion.reanimated.keyboard;
+
+import android.app.Dialog;
+import android.content.DialogInterface;
+import android.view.Window;
+import android.view.WindowManager;
+import com.facebook.react.bridge.ReactApplicationContext;
+import java.lang.ref.WeakReference;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Stack;
+
+public class ModalActivityManager {
+  public static String OPEN_MODAL_EVENT_NAME = "topShow";
+
+  private Integer insetManagersCount = 0;
+
+  private boolean mIsStatusBarTranslucent = false;
+
+  private boolean isObserving = false;
+
+  private final WeakReference<ReactApplicationContext> mReactApplicationContext;
+
+  private final NotifyAboutKeyboardChangeFunction mNotifyAboutKeyboardChange;
+
+  private final Keyboard mKeyboard;
+
+  private final Stack<Window> modalsStack = new Stack<>();
+
+  private final HashMap<Integer, WindowsInsetsManager> insetsManagersMap = new HashMap<>();
+
+  public ModalActivityManager(
+      WeakReference<ReactApplicationContext> reactContext,
+      Keyboard keyboard,
+      NotifyAboutKeyboardChangeFunction notifyAboutKeyboardChange) {
+    mNotifyAboutKeyboardChange = notifyAboutKeyboardChange;
+    mReactApplicationContext = reactContext;
+    mKeyboard = keyboard;
+
+    createInsetManager(); // Create inset manager for main window
```

Comment:
Let's relocate it to `KeyboardAnimationManager`, as it seems counterintuitive for `ModalActivityManager` to create an observer for the main window as well.

Reply:
Fixed in: https://github.com/software-mansion/react-native-reanimated/pull/5916/commits/25c1a9439d6e2845654e82dcadbb7da0baede236